### ISSUE TITLE
no-unused-vars option to allow unused vars in object destructuring with rest property

### ISSUE THREAD

**Issue by avionbg** [sweepline/eslint-plugin-unused-imports#41]
Issue is described in original no-unused-vars repository: https://github.com/eslint/eslint/issues/4880

no-unused-vars should have an option to ignore "skipped" object properties when destructuring.

**Comment by sweepline** [comment]
This already works, all the options are passed to the underlying rule. You can do like this.
```js
		"@typescript-eslint/no-unused-vars": "off",
		"unused-imports/no-unused-imports": "error",
		"unused-imports/no-unused-vars": [
			"warn",
			{
				ignoreRestSiblings: false,
			},
		],
```
You are probably running an outdated version of eslint or typescript-eslint.

**Comment by christophsturm** [comment]
I have exactly the same problem as the OP:

this works: 
```
	rules: {
		'no-constant-condition': ['error', { checkLoops: false }],
		'@typescript-eslint/no-unused-vars': [
			'warn',
			{
				args: 'all',
				argsIgnorePattern: '^_',
				caughtErrors: 'all',
				caughtErrorsIgnorePattern: '^_',
				destructuredArrayIgnorePattern: '^_',
				varsIgnorePattern: '^_',
				ignoreRestSiblings: true
			}
		]
	}
```
but when i change it to this, it does not work:
```
	rules: {
		'no-constant-condition': ['error', { checkLoops: false }],
		'no-unused-vars': 'off',
		'unused-imports/no-unused-imports': 'error',
		'unused-imports/no-unused-vars': [
			'warn',
			{
				args: 'all',
				argsIgnorePattern: '^_',
				caughtErrors: 'all',
				caughtErrorsIgnorePattern: '^_',
				destructuredArrayIgnorePattern: '^_',
				varsIgnorePattern: '^_',
				ignoreRestSiblings: true
			}
		]

	}
```
 
```  eslint-plugin-unused-imports:
    specifier: ^3.1.0
    version: 3.1.0(@typescript-eslint/eslint-plugin@7.3.1)(eslint@8.57.0)
```

**Comment by sweepline** [comment]
It is simply not true. Can you give me a minimal reproducible example of a file where it does not work.

```ts
const x = { a: 1, b: 2, c: 3 };
const { a, ...rest } = x;
console.log(rest);
```
This example reports not errors with the following config file.
```js
module.exports = {
	parser: "@typescript-eslint/parser",
	parserOptions: {
		ecmaVersion: 2022,
		sourceType: "module",
	},
	plugins: ["@typescript-eslint", "unused-imports"],
	rules: {
		"@typescript-eslint/no-unused-vars": [
			"warn",
			{
				vars: "all",
				varsIgnorePattern: "^_",
				args: "after-used",
				argsIgnorePattern: "^_",
				ignoreRestSiblings: true,
			},
		],
		"unused-imports/no-unused-vars": [
			"warn",
			{
				vars: "all",
				varsIgnorePattern: "^_",
				args: "after-used",
				argsIgnorePattern: "^_",
				ignoreRestSiblings: true,
			},
		],
	},
};
```

**Comment by christophsturm** [comment]
it seems the problem was in my config file, it specified just 		`'no-unused-vars': 'off',` if i change it to `'@typescript-eslint/no-unused-vars': 'off',` it works.
